### PR TITLE
Add -D: duration option to check

### PR DESCRIPTION
**Pull request by kbara**

### CODE REVIEW

File: src/program/lwaftr/check/README [hunk 1]
```diff
@@ -5,8 +5,10 @@ Usage: check [-r] CONF V4-IN.PCAP V6-IN.PCAP V4-OUT.PCAP V6-OUT.PCAP [COUNTERS.L
                              Print usage information.
   -r, --regen
                              Regenerate counter files, instead of checking.
-  --on-a-stick
+  --on-a-stickef
```
Comment (teknico) [inline src/program/lwaftr/check/README]: This looks like something that got away. :-)

File: src/program/lwaftr/check/check.lua [hunk 2]
```diff
@@ -117,7 +120,7 @@ function run(args)
          validate_diff(counters_diff, req_counters)
       end
    else
-      engine.main({duration=0.10})
+      engine.main({duration=opts.duration or 0.10})
```
Comment (dpino) [inline src/program/lwaftr/check/check.lua]: It's missing to set a duration on the if branch. I think is better to set the a default value for duration in `parse_args` if not defined and make `engine.main({duration=opts.duration})` in both branches.

**Comment by kbara** [comment]
@dpino happy with this version?

File: src/program/lwaftr/check/check.lua [hunk 3]
```diff
@@ -28,9 +28,13 @@ function parse_args (args)
...
    end
-   args = lib.dogetopt(args, handlers, "hr",
-      { help="h", regen="r", ["on-a-stick"] = 0 })
+   handlers.D = function(dur)
+      opts["duration"] = tonumber(dur)
+   end
+   args = lib.dogetopt(args, handlers, "hrD:",
+      { help="h", regen="r", duration="D", ["on-a-stick"] = 0 })
    if #args ~= 5 and #args ~= 6 then show_usage(1) end
+   if not opts["duration"] then opts[duration] = 10 end
```
Comment (dpino) [inline src/program/lwaftr/check/check.lua]: There's a typo, should be 0.10 instead.

Comment (dpino) [reply src/program/lwaftr/check/check.lua]: And the quotes are missing, that's why the CI failed.

```
program/lwaftr/check/check.lua:37: variable 'duration' is not declared
```

File: src/program/lwaftr/check/README (hunk 1 above)
Comment (kbara) [reply src/program/lwaftr/check/README]: Yep, good catch.

**Comment by dpino** [comment]
Sorry I think I didn't explain myself properly. The issue with the patch is that it only sets the duration for the else branch, while the if branch still has a fixed duration of 0.10. I think the -D knob should set duration always.

https://github.com/Igalia/snabb/pull/464/files#diff-ed55dfa416547983ea5b3fb5480037c2R114

**Review by dpino — APPROVED** [review]

**Pull request merged**